### PR TITLE
feat(api): D1 attachment index, phase 1 (dual-write) (#934)

### PR DESCRIPTION
**Pull request by zachdunn**
Phase 1 of the D1 attachment index from #934. Write-only: a new `github_attachments` table is populated from every attachment write path, and nothing reads it yet. `gatherCommentBody` is untouched, so this PR changes no user-visible behavior; it exists so Phase 2 can shadow-compare the index against the R2 fan-out before switching the comment sync over.

## What's in here

**Schema.** `apps/api/migrations/20260903120000_github_attachments.sql`: one row per attachment object, PK `(workspace, object_key)`, partial index on `(workspace, repo, kind, num, object_key) WHERE detached_at IS NULL` for the eventual hot read, plus prefix-id and repo indexes for rotation and reconcile. `kind` uses the `GhTarget` spelling (`pull` | `issues`). Auto-applies on merge via the D1 migrations workflow.

**Writer module.** `apps/api/src/github-attachment-index.ts`: `parseAttachmentKey`, `recordAttachment` (upsert; keeps `created_at`; a plain `put` re-record leaves an existing `detached_at` alone so the index agrees with `gh.detached` and the adopt ledger, while attach / promote / adopt clear it), `detachAttachment` / `reattachAttachment`, single, chunked-batch and whole-workspace deletes, `rekeyAttachment` (destination-first wipe, then update), and never-throwing `*Safe` variants that log a JSON line and move on. Index writes are best-effort in the same sense as `recordUsageSafe`: a D1 failure costs a stale index row, never a failed upload.

**Trust boundary.** Every row's `repo` comes from the final object key (plain `gh/<owner>/<name>/…` keys) or from the `github_private_prefixes` row that minted the prefix id (private keys, via the new `repoForPrefixId`). It is never read from `gh.*` file_metadata, which is client-settable. A private key whose prefix id has no owning row writes no row. Tests pin that a client-supplied `gh.repo` / `gh.ref` naming another repo does not influence the recorded repo.

**Write sites.**

| Path | Effect |
| --- | --- |
| `putObject` (choke point for REST PUT, idempotent PUT, reconcile, attach, promote, rotation) | upsert, `source: "put"` |
| `attachExistingObject` | re-record with `source: "attach"` and the caller-resolved repo |
| `promoteBranchAttachments` | re-record with `source: "promote"`; the staged original gets no row |
| link adoption | `source: "adopt"`; detach / re-attach flip `detached_at` alongside `gh.detached` |
| private-prefix rotation | `rekeyAttachment` next to the existing `file_metadata` re-key |
| `deleteObject`, retention purge, workspace teardown | rows removed alongside the metadata cleanup |
| GitHub-native ingest | deliberately no rows; guard test asserts it |

**Tests.** A sqlite-backed suite for the module (parse, upsert, detach, chunked delete past the 100-parameter cap, re-key with an occupied destination, safe wrappers), a hand-rolled fake table wired into the shared test D1 so the route/webhook/adopt/ingest suites keep running, and a wiring test per write site. `apps/api`: 2611 tests; monorepo `pnpm test` green.

## Not in this PR

Phase 2 (shadow compare behind a flag), Phase 3 (read from the index), the reconcile job, and the backfill. Plan and rollout are in the #934 comment thread.

Deferred from review into the Phase 2 checklist, all performance shape rather than correctness: attach/adopt currently write the row twice (`putObject` then the caller re-records with its own `source`); the `putObject` index write is awaited serially after `recordContentHash`; rotation and retention call the index for keys that can never have a row. Also deferred: the two-statement re-key is not a `db.batch()`, and `kind` has no schema CHECK.

## Reviewer notes

- The migration is additive; no existing table or query changes.
- Per-write cost is one upsert (plus one indexed `github_private_prefixes` lookup for private keys). No new R2 calls anywhere.

## Summary by CodeRabbit

* **New Features**
  * Added reliable tracking for GitHub issue and pull request attachments.
  * Attachment records now stay synchronized when files are uploaded, copied, adopted, moved, rotated, or removed.
  * Added server-side repository resolution to improve attachment ownership accuracy.

* **Bug Fixes**
  * Prevented stale attachment records after retention cleanup or workspace deletion.
  * Improved handling when detached attachments are referenced again.

* **Tests**
  * Added comprehensive coverage for attachment tracking, lifecycle changes, cleanup, rotation, and security-related edge cases.

### CODE REVIEW

File: apps/api/src/files-core.ts [hunk 1]
```diff
@@ -1270,6 +1291,11 @@ export async function deleteObject(
   const configs = await storageConfigs(env, ws);
   const { hits, failures } = await deleteFromEveryLane(configs, key);
   await deleteFileMetadata(dbFor(env), workspaceName, key);
+  // The attachment index (issue #934) is keyed by (workspace, object_key)
+  // just like file_metadata, so it is cleaned up in exactly the same place.
+  // Best-effort: a stale row costs a broken image in a managed comment
+  // until reconcile, never a failed delete.
+  await deleteAttachmentSafe(dbFor(env), workspaceName, key);
```
[the remark this answers is not in the document]
Comment (zachdunn) [reply apps/api/src/files-core.ts]: Agreed and fixed in d84020cd: `deleteAttachmentSafe` now runs after the `failures.length` check, so a lane that still holds the object keeps its index row. Added a test that a throwing lane delete rejects and leaves the row in place.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
